src/main/java/org/apache/commons/csv/CSVFormat.java

```diff
@@ -2280,6 +2279,27 @@ private void validate() throws IllegalArgumentException {
         }
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        // check whether field duplicateHeaderMode (as of 1.10.0) was found in serialized form
+        // if not, set from boolean allowDuplicateHeaderNames
+        if (duplicateHeaderMode == null) {
+            setFieldValue("duplicateHeaderMode",
+                    allowDuplicateHeaderNames ? DuplicateHeaderMode.ALLOW_ALL : DuplicateHeaderMode.DISALLOW);
+        }
+        return this;
+    }
+
+    private void setFieldValue(String fieldName, Object value) {

```

Comment:
Use `final` where you can.

Reply:
In Java private methods are non-virtual.
I will put `final` on methods parameters.

Reply:
> In Java private methods are non-virtual. I will put `final` on methods parameters.

Of course, this comment applies only to parameters, just look at the rest of the code base ;-)